src/shared/otterizecloud/graphqlclient/errors.go

```diff
@@ -0,0 +1,34 @@
+package graphqlclient
+
+import (
+	"errors"
+	"github.com/vektah/gqlparser/v2/gqlerror"
+)
+
+func GetGraphQLUserError(err error) (UserErrorType, string, bool) {
+	errList := &(gqlerror.List{})
+	if !errors.As(err, errList) {
+		return "", "", false
+	}
+	gqlErr := &gqlerror.Error{}
+	if !errList.As(&gqlErr) {
+		return "", "", false
+	}
+	errType, ok := gqlErr.Extensions["errorType"].(string)
+	if !ok || !UserErrorType(errType).IsValid() {
```

Comment:
I don't think it requires an error. If errType is not valid, it's probably because the graphQL is not current. It makes more sense to use `error.Error()` than throwing another kind of error.